dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/ZipFileArtifactNotifier.java

```diff
@@ -644,11 +648,18 @@ public void onChange(
 
         boolean isUpdate = !updatedFiles.isEmpty();
         if ( addedFiles.isEmpty() && !isUpdate && removedFiles.isEmpty() ) {
-            return;
+            return; // Ignore completely null updates
         }
-        validateNotification(addedFiles, updatedFiles, removedFiles);
 
-        notifyAllListeners(isUpdate);
+        // The net is to allow updates alone or removals alone.
+
+        String validationMessage = validateNotification(addedFiles, updatedFiles, removedFiles);
+        if ( validationMessage != null ) {
+            // TODO: Need to understand what is happening here.
+            // Tr.warning(tc,  "Unexpected notification on [ " + rootContainer + " ]: " + validationMessage);
```

Comment:
should a debug trace go here?  otherwise the validationMessage won't be traced since that method is marked `@Trivial`